app/models/name.rb

```diff
@@ -2219,12 +2102,64 @@ def self.new_name(params)
     result
   end
 
+  # Make a Name instance from a ParsedName
+  # Used by NameController#create_new_name
+  # Returns a Name instance, *UNSAVED*!!
+  def self.new_name_from_parsed_name(parsed_name)
+    new_name(parsed_name.params)
+  end
+
+  # Return extant Names matching a desired new Name
+  # Used by NameController#create_name
+  def self.names_matching_desired_new_name(parsed_name)
+    # unauthored ParsedName matches Names with or w/o authors
+    if parsed_name.author.empty?
```

Comment:
And you are also right about (a) being easy to change and (b) how to change NameController. (I don't even have to change any tests -- I didn't test creating an unauthored group when an authored one exists. If we're going to change it, I should add a test.)